ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ExecuteQueryResponseHandler.java

```diff
@@ -59,30 +58,34 @@
 
   @Override
   protected void handleMetadata(final QueryResponseMetadata queryResponseMetadata) {
-    cf.queryID().complete(queryResponseMetadata.queryId);
+    final String queryId = queryResponseMetadata.queryId;
+
+    cf.queryID().complete(queryId);
+
     columnNames = queryResponseMetadata.columnNames;
-    columnTypes = RowUtil.columnTypesFromStrings(queryResponseMetadata.columnTypes);
+
+    final List<String> columnTypesString = queryResponseMetadata.columnTypes;
+
+    columnTypes = RowUtil.columnTypesFromStrings(columnTypesString);
+
     columnNameToIndex = RowUtil.valueToIndexMap(columnNames);
   }
 
   @Override
-  protected void handleRow(final Buffer buff) {
-    final Row row;
-    final Object json = buff.toJson();
-
-    if (json instanceof JsonObject) {
-      final JsonObject jsonObject = (JsonObject) json;
+  protected void handleRow(final JsonNode rowNode) {
+    if (rowNode.has(CONSISTENCY_TOKEN_KEY)) {
       // This is the serialized consistency vector
       // Don't add it to the result list since the user should not see it
-      if (jsonObject.getMap() != null && jsonObject.getMap().containsKey("consistencyToken")) {
-        log.info("Response contains consistency vector " + jsonObject);
-        serializedConsistencyVector.set((String) ((JsonObject) json).getMap().get(
-            "consistencyToken"));
-      } else {
-        throw new RuntimeException("Could not decode JSON, expected consistency toke: " + json);
-      }
-    } else  if (json instanceof JsonArray) {
-      final JsonArray values = new JsonArray(buff);
+
+      final String consistencyTokenString = rowNode
+              .get(CONSISTENCY_TOKEN_KEY)
```

Comment:
Extra get?

Reply:
no. the `ConsistencyToken` object has a `consistencyToken`. instance variable